crates/provider/src/provider.rs

```diff
@@ -995,10 +996,65 @@ impl<N: Network, T: Transport + Clone> Provider<N, T> for RootProvider<N, T> {
     }
 }
 
+/// Admin namespace rpc interface that gives access to several non-standard RPC methods.
+#[allow(unused, unreachable_pub)]
+pub trait AdminApi {
```

Comment:
Please use `async_trait` to make it object-safe, see above.

Reply:
I used `impl_future!` for every methods, shouldn't it be enough?

Reply:
hmm, that's a valid point although since we can box the provider and impl the trait for any provider this trait doesn't really need to be object safe, but fine with using async_trait here as well.

Reply:
Done.